Filter public link share data

settermjd opened the pull request:
## Description

Filter data passed to the script that emails a user about a public link share.

## Related Issue

https://github.com/owncloud/security-tracker/issues/255

## Motivation and Context

It closes an HTML injection vulnerability.

## How Has This Been Tested?

There is a limited ability to suitably test `/ocs/ajax/share.php`. Given that, I created tests for the class that I developed. Then, I manually tested the changes via the browser.

## Screenshots (if appropriate):

Relevant screenshots are contained in the related issue.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

settermjd committed:
Add class to filter link share email script parameters

The PHP and related JavaScript code for creating a public link share
email don't (at least as far as I can tell) do any form of validation
nor filtering of the four parameters (link, file, to address, and
expiration date) which the share script needs to create the email.

Given that, this class was created to provide an elementary form of
filtering to help avert the HTML injection security vulnerability
identified in https://github.com/owncloud/security-tracker/issues/255.

tomneedham commented:
We should validate the email addresses with https://github.com/owncloud/core/blob/master/lib/private/Mail/Mailer.php#L118

settermjd commented:
I deliberately avoided adding validation, as I'm not that familiar with the rest of the codebase, and felt that that may lead down a dark rabbit hole. However, I'm more than happy to add that in a separate PR.

settermjd committed:
Filter user input to the public link notification email

This commit doesn’t perform any validation, just basic filtering, to 
prevent malicious input from being used in the email.

settermjd commented:
I made minor amendments. Any input @tomneedham @PVince81 @DeepDiver1975 would be appreciated.

PVince81 commented on composer.json:
```diff
@@ -58,6 +58,8 @@
         "sabre/dav": "^3.2",
-        "deepdiver/zipstreamer": "^1.1"
+        "deepdiver/zipstreamer": "^1.1",
+        "zendframework/zend-inputfilter": "^2.8",
```
@DeepDiver1975 are we ok to bring in these new libs ?

DeepDiver1975 replied:
license is  BSD-3-Clause - need to chat with @hodyroff about this

hodyroff replied:
License is good, need to add to our list.

PVince81 commented on lib/private/Share/Filters/MailNotificationFilter.php:
```diff
@@ -0,0 +1,82 @@
+
+use Zend\InputFilter\Factory as InputFilterFactory;
+
+class MailNotificationFilter {
```
PHPDoc, what is this class about ?

settermjd replied:
Will get that updated.

settermjd commented:
@PVince81, file and class dockblock comments added. Thanks for spotting that.

DeepDiver1975 commented:
@ownclouders rebase

ownclouders commented:
Hey! I'm [GitMate.io](https://gitmate.io)! This pull request is being rebased automatically. Please **DO NOT** push while rebase is in progress or your changes would be lost permanently :warning:

ownclouders commented:
Automated rebase failed! Please rebase your pull request manually via the command line.

Reason:
```
Command: git rebase base/master

Exit Code: 128

Cause:

error: Failed to merge in the changes.


```

PVince81 commented:
@settermjd can you rebase ? Seems the bot is not able today 😄 (job security for professional rebasers FTW)

settermjd commented:
@PVince81 can do.

settermjd commented:
@PVince81, merged with master.

DeepDiver1975 commented on composer.lock:
```diff
@@ -2246,16 +2326,16 @@
         {
             "name": "sabre/vobject",
-            "version": "4.1.3",
+            "version": "4.1.4",
```
this should not have happened

DeepDiver1975 commented on composer.lock:
```diff
@@ -3269,12 +3644,12 @@
                 "type": "git",
                 "url": "https://github.com/minkphp/MinkSelenium2Driver.git",
-                "reference": "46404b04c761feff4feab4d8d75957225e21a73c"
+                "reference": "93474c65a2a7bf959200ab5f7a14cc450645c185"
```
same - please do not run composer update

settermjd replied:
Fair call. Will change.

settermjd commented:
@DeepDiver1975; file revised.

settermjd committed:
Add required dependencies

DeepDiver1975 commented:
I assume we want to backport this ... @settermjd

DeepDiver1975 commented:
... will take care

DeepDiver1975 commented:
backport open
